tests/cuda/test_kernel_semantic.py

```diff
@@ -0,0 +1,177 @@
+                                    INVALID_KERNEL_CALL_BP_GRID,
+                                    INVALID_KERNEL_LAUNCH_CONFIG_HEIGHT,
+                                    INVALID_KERNEL_LAUNCH_CONFIG_LOW)
+
+
+@pytest.mark.cuda
+def test_invalid_block_number():
+    def invalid_block_number():
+        @kernel
+        def kernel_call():
+            pass
+
+        blocks_per_grid = 1.0
+        threads_per_block = 1
+        kernel_call[blocks_per_grid, threads_per_block]()
+
+    errors = Errors()
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(invalid_block_number, language="cuda")
+
+    assert errors.has_errors()
+
+    assert errors.num_messages() == 1
+
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.funcdef == 'kernel_call'
+    assert INVALID_KERNEL_CALL_BP_GRID == error_info.message
+
+
+@pytest.mark.cuda
+def test_invalid_thread_per_block():
+    def invalid_thread_per_block():
+        @kernel
+        def kernel_call():
+            pass
+
+        blocks_per_grid = 1
+        threads_per_block = 1.0
+        kernel_call[blocks_per_grid, threads_per_block]()
+
+    errors = Errors()
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(invalid_thread_per_block, language="cuda")
+    assert errors.has_errors()
+    assert errors.num_messages() == 1
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.funcdef == 'kernel_call'
+    assert INVALID_KERNEL_CALL_TP_BLOCK == error_info.message
+
+
+@pytest.mark.cuda
+def test_invalid_launch_config_height():
+    def invalid_launch_config_height():
+        @kernel
+        def kernel_call():
+            pass
+
+        blocks_per_grid = 1
+        threads_per_block = 1
+        third_param = 1
+        kernel_call[blocks_per_grid, threads_per_block, third_param]()
+
+    errors = Errors()
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(invalid_launch_config_height, language="cuda")
+
+    assert errors.has_errors()
+    assert errors.num_messages() == 1
+
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.funcdef == 'kernel_call'
+    assert INVALID_KERNEL_LAUNCH_CONFIG_HEIGHT == error_info.message
+
+
+@pytest.mark.cuda
+def test_invalid_launch_config_low():
+    def invalid_launch_config_low():
+        @kernel
+        def kernel_call():
+            pass
+
+        blocks_per_grid = 1
+        kernel_call[blocks_per_grid]()
+
+    errors = Errors()
+
+    with pytest.raises(PyccelSemanticError):
+        epyccel(invalid_launch_config_low, language="cuda")
+
+    assert errors.has_errors()
+    assert errors.num_messages() == 1
+
+    error_info = [*errors.error_info_map.values()][0][0]
+    assert error_info.symbol.funcdef == 'kernel_call'
+    assert INVALID_KERNEL_LAUNCH_CONFIG_LOW == error_info.message
+
+
+@pytest.mark.cuda
+def test_invalid_arguments_for_cuda_language():
```

Comment:
What do you think about renaming this test to `test_invalid_arguments_for_kernel_call`?

Reply:
fixed. 26cf144